src/HttpRequest.php

```diff
@@ -279,4 +282,19 @@ private function getServerVariable($key)
 
         return $this->server[$key];
     }
+
+    private function parseInputParameters()
+    {
+        if (isset($this->server['REQUEST_METHOD']) && isset($this->server['HTTP_CONTENT_TYPE']) &&
+            ($this->getMethod() == 'PUT' || $this->getMethod() == 'PATCH'))
+        {
+            if ($this->server['HTTP_CONTENT_TYPE'] == 'application/x-www-form-urlencoded') {
+                parse_str($this->inputStream, $this->postParameters);
+            }
+            else if ($this->server['HTTP_CONTENT_TYPE'] == 'application/json') {
+                $this->postParameters = json_decode($this->inputStream, true);
+            }
```

Comment:
i agree, i kinda made it in hindsight because i was annoyed i couldn't find a small and nice package that does evereything i need :D I really like the package, however Ideally my wish is to make it great for APIs too and to do that we need to be able to handle JSON requests and responses and all HTTP verbs as well. I belive we could do that while keeping all of the current functionallity and without bloating it too much. Let me know if you would like to see this package go that way and i'll spend some more time thinking about how to do it exactly.